internal/observability/graph.go

```diff
@@ -199,6 +201,12 @@ func newEdge(logger *zap.Logger, meter *metrics.Scope, req *transport.Request, d
 		"rpc_type":         rpcType.String(),
 	}
 
+	if metricTagsBlocklist != nil {
```

Comment:
No need of checking nil here, golang can handle `for _, filteredKey := range nil`